pkg/apis/service/validation/validation.go

```diff
@@ -86,12 +88,39 @@ func validateIssuers(cluster *controller.Cluster, issuers []service.IssuerConfig
 		if issuer.RequestsPerDayQuota != nil && *issuer.RequestsPerDayQuota < 1 {
 			allErrs = append(allErrs, field.Invalid(indexFldPath.Child("requestsPerDayQuota"), *issuer.RequestsPerDayQuota, "must be >= 1"))
 		}
+		if len(issuer.PrecheckNameservers) > 0 {
+			for j, server := range issuer.PrecheckNameservers {
+				if err := validateNameserver(server); err != nil {
+					allErrs = append(allErrs, field.Invalid(indexFldPath.Child("precheckNameservers").Index(j), server, err.Error()))
+				}
+			}
+		}
 		names.Insert(issuer.Name)
 	}
 
 	return allErrs
 }
 
+func validateNameserver(server string) error {
+	host, port, err := net.SplitHostPort(server)
+	if err != nil {
```

Comment:
What happened if one configures an invalid domain/name without a port number? IIUC, then the `host = server` and the validation doesn't fail in this case (see also my comment for `validation_test.go`).

Reply:
As said, port 53 is assumed if no port is given.